astropy/coordinates/builtin_frames/skyoffset.py

```diff
@@ -177,3 +177,10 @@ def represent_as(self, base, s='base', in_frame_units=False):
         data = super().represent_as(base, s, in_frame_units=in_frame_units)
         self._set_skyoffset_data_lon_wrap_angle(data)
         return data
+
+    def __reduce__(self):
+        return (_skyoffset_reducer, ({'origin': self.origin}, ), self.__dict__)
+
+
+def _skyoffset_reducer(kw):
+    return SkyOffsetFrame(**kw)
```

Comment:
I think this will execute `__init__`, which in principle is not wanted. I think it should be
```
    return SkyOffsetFrame.__new__(SkyOffsetFrame, **kw)
```

Or, if you go with my suggestion above,
```
def _skyoffset_reducer(origin):
    return SkyOffsetFrame.__new__(SkyOffsetFrame, origin=origin)
```